compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -2025,11 +2030,24 @@ private BType defineSymbolForCyclicTypeDefinition(BLangTypeDefinition typeDef, S
                         newTypeDefName, env.enclPkg.symbol.pkgID, newTypeNode, env.scope.owner,
                         typeDef.name.pos, SOURCE);
                 break;
-            default:
+            case INTERSECTION_TYPE_NODE:
+                newTypeNode = new BIntersectionType(null, new LinkedHashSet<>(),
+                        (IntersectableReferenceType) symTable.anyType);
```

Comment:
How are we just using `any` type here as the effective type?

Reply:
We need to use a relevant type for effective type in recursive cyclic intersection types.
ex: 
```
type T readonly & (A|B);
type A readonly & [T];
type B readonly & [T];

type C readonly & [D];
type D readonly & [C]
```

For this type there is no effective type ryt?
What will be the most suitable type?